src/pages/students/edit-student/EditStudent.tsx

```diff
@@ -0,0 +1,135 @@
+import StudentData, { type StudentDataProps } from "../forms/studentDataForm/StudentData"
+import PaymentData, { type PaymentDataProps } from "../forms/paymentDataForm/PaymentData"
+import { useGetStudentByIdQuery, useUpdateStudentMutation } from "../../../app/services/StudentService"
+import useAuthentication from "../../../hooks/useAuthentication";
+import { useRef, type JSX } from "react";
+//import PlanData, { type PlanDataProps } from "../forms/planDataForm/PlanData";
+import { useLocation, useNavigate } from "react-router";
+import { useParams } from "react-router-dom";
+import { MainContainer, TitleContainer, Title, ButtonsContainer, Button, FormContainer } from "./EditStudent.styles";
+import { formatDateToDash, formatDateToISO } from "../../../utils/DateFormatter";
+import toast from "react-hot-toast";
+import type { UpdateStudentRequest } from "../../../app/types/requests/UpdateStudentRequest.type";
+import BackIcon from "../../../assets/back-icon.svg";
+
+
+function EditStudent() {
+
+    const { userId } = useAuthentication();
+
+    const formStudentDataRef = useRef<any>(undefined);
+
+    const formPaymentDataRef = useRef<any>(undefined);
+
+    const { numberOfStep } = useParams();
+
+    const { studentId } = useLocation().state;
```

Comment:
Obtener el ID de esta manera tiene un bug, y es que el usuario puede ingresar a la seccion directamente desde la URL, es decir, escribir la URL directamente ocasionando lo siguiente

<img width="2269" height="1279" alt="Image" src="https://github.com/user-attachments/assets/fcad4063-6ff9-4b4e-b1cb-67cd60259589" />

Te queda la página en blanco porque al escribir directamente la URL, no se está pasando el state, por lo que se rompe todo. Una manera de arreglar esto es primero verificar si el state existe, y en base a eso, obtener el studentId


```suggestion
    const state = useLocation().state;
    if (!state) {
        return <p>Hubo un error al cargar la página. Intente nuevamente</p>;
    }
    const { studentId } = state as { studentId: string };
```

De esta manera, si bien el usuario no va a poder editar los datos, al menos va a poder seguir navegando en la aplicacion

<img width="2082" height="918" alt="Image" src="https://github.com/user-attachments/assets/13896acb-7dab-4095-9b16-2a5771defbc2" />